### DISCUSSION TITLE
[BUG] bundle size

### DISCUSSION THREAD

**Discussion by spice2445** [family/connectkit#369] [Ideas]
## Describe the bug
When I install the library, the bandle size increases by 1mb. Although I use only three functions from the library:
import { getDefaultConfig, ConnectKitProvider, useModal } from 'connectkit'
As I understand it there is no tree shaking in the library, which is very unfortunate(

## To reproduce
npm i connectkit
try to import { getDefaultConfig, ConnectKitProvider, useModal } from 'connectkit'
and see the bundle size
it increases by 1mb

## Additional context
Using ConnectKit 1.7.2

**Comment by wottpal** [comment]
Same issue

**Comment by spice2445** [comment]
Any word on if this issue? We're trying to decide if we want to fork and tackle (less likely) or switch to a different wallet library (more likely).

**Comment by lochie** [comment]
Are you calculating the bundle size using an extension such as [Import Cost](https://marketplace.visualstudio.com/items?itemName=wix.vscode-import-cost)? If so I've found that that extension in particular doesn't calculate this package correctly.

You're welcome to look into if there's any obvious tree-shaking improvements that could be done, but a lot of the overhead is from the required peer dependencies, such as WalletConnect SDK and CoinbaseWallet SDK (imported from wagmi/viem).

**Reply by lochie** [reply]
Converted this to a discussion as this isn't exactly a bug, but more of a requested improvement.
